### PR TITLE
Add support for availability sets

### PR DESCRIPTION
**Pull request by liwonMS**

### CODE REVIEW

File: services/compute/virtualmachine/mockclient.go [hunk 1]
```diff
@@ -0,0 +1,434 @@
...
+	"github.com/microsoft/moc/pkg/auth"
+	"github.com/microsoft/moc/pkg/errors"
+)
+
+type VirtualMachineMockClient struct {
+	vmClient      VirtualMachineClient
+	avsetInternal computeavset.Service
+}
+
+func NewVirtualMachineMockClient(cloudFQDN string, authorizer auth.Authorizer) (*VirtualMachineMockClient, error) {
```
Comment (msftpgayam) [inline services/compute/virtualmachine/mockclient.go:24]: Where is this client used?

File: services/compute/compute.go [hunk 2]
```diff
@@ -1160,3 +1162,33 @@ type ProxyConfiguration struct {
...
+	// Statuses - Statuses
+	Statuses map[string]*string `json:"statuses"`
+	// Fault domain count
+	PlatformFaultDomainCount *int32 `json:"platformFaultDomainCount,omitempty"`
+	// VMs
+	VirtualMachines []*CloudSubResource
+}
+
+// CloudSubResource describes a resoruce reference setting for a virtual machine
+type CloudSubResource struct {
```
Comment (msftpgayam) [inline services/compute/compute.go:1189]: nit: can we call it `AvailabilitySetProfile` to keep it consistent?

File: services/compute/virtualmachine/mockclient.go (hunk 1 above)
Comment (liwonMS) [reply services/compute/virtualmachine/mockclient.go:24]: will remove, this is for akshybrid to test in prrallel.

**Review by msftpgayam — DISMISSED** [review]

**Review by msftpgayam — APPROVED** [review]

**Pull request closed without merging**